main.tf

```diff
@@ -1,6 +1,6 @@
 locals {
   master_password      = var.create_db_instance && var.create_random_password ? random_password.master_password[0].result : var.password
-  db_subnet_group_name = var.replicate_source_db != null ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id)
+  db_subnet_group_name = var.replicate_source_db != null ? null : coalesce(var.db_subnet_group_name, module.db_subnet_group.db_subnet_group_id, var.identifier)
```

Comment:
Btw, we have some similar workaround for `coalesce()` in other modules where it was not possible to import resources. Eg: https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/iam.tf#L14